src/js/events/MesosSummaryActions.js

```diff
@@ -1,58 +1,95 @@
-import ActionTypes from '../constants/ActionTypes';
+import {
+  REQUEST_MESOS_HISTORY_SUCCESS,
+  REQUEST_MESOS_HISTORY_ONGOING,
+  REQUEST_MESOS_SUMMARY_SUCCESS,
+  REQUEST_MESOS_SUMMARY_ERROR,
+  REQUEST_MESOS_SUMMARY_ONGOING
+} from '../constants/ActionTypes';
+
 var AppDispatcher = require('./AppDispatcher');
 var Config = require('../config/Config');
 var RequestUtil = require('../utils/RequestUtil');
 var _historyServiceOnline = true;
 
-function getStateUrl(timeScale) {
-  timeScale = timeScale || 'last';
-  if (_historyServiceOnline) {
-    return `${Config.historyServer}/dcos-history-service/history/${timeScale}`;
-  } else {
-    return `${Config.rootUrl}/mesos/master/state-summary`;
+function testHistoryOnline() {
+  RequestUtil.json({
+    url: `${Config.historyServer}/dcos-history-service/history/last`,
+    success: function () {
+      _historyServiceOnline = true;
+    },
+    error: function () {
+      setTimeout(testHistoryOnline, Config.testHistoryInterval);
+    }
+  });
+}
+
+function getHistory(resolve, reject, timeScale = 'last') {
+  let url = `${Config.historyServer}/dcos-history-service/history/${timeScale}`;
+  let successEventType = REQUEST_MESOS_SUMMARY_SUCCESS;
+
+  if (timeScale === 'minute') {
+    successEventType = REQUEST_MESOS_HISTORY_SUCCESS;
   }
+
+  RequestUtil.json({
+    url,
+    success: function (response) {
+      AppDispatcher.handleServerAction({
+        type: successEventType,
+        data: response
+      });
+      resolve();
+    },
+    error: function () {
```

Comment:
It's my opinion that we still dispatch an action, we do this in several places where we don't need it, but still do it.
If you disagree at least remove it from the action types constant file. `REQUEST_MESOS_HISTORY_ERROR`